lib/services/geolocation_service.dart

```diff
@@ -42,28 +43,57 @@ class GeoLocationService {
       permission = await _geoLocator.requestPermission();
       if (permission == LocationPermission.denied) {
         // Permissions are denied after requesting
-        throw Exception("Location permissions are denied");
+        return Exception("Location permissions are denied");
       }
     }
 
     if (permission == LocationPermission.deniedForever) {
       // Permissions are denied forever
       // TODO : Handle this case to notify the user that the location permissions are denied forever
-      throw Exception(
+      return Exception(
         "Location permissions are permanently denied, we cannot request permissions.",
       );
     }
+    return null;
+  }
 
+  /// This method will retrieve the current position of the user.
+  Future<GeoPoint> getCurrentPosition() async {
+    //check if location services are enabled
+    final state = await checkLocationServices();
+    if (state != null) {
+      throw state;
+    }
     final position = await _geoLocator.getCurrentPosition(
       locationSettings: locationSettings,
     );
 
     return GeoPoint(position.latitude, position.longitude);
   }
+
+  Stream<GeoPoint> getPositionStream() async* {
+    //check if location services are enabled
+    final state = await checkLocationServices();
+    if (state != null) {
+      throw state;
+    }
+
+    await for (final position in _geoLocator.getPositionStream(
+      locationSettings: locationSettings,
+    )) {
+      yield GeoPoint(position.latitude, position.longitude);
+    }
+  }
 }
 
 final geoLocationServiceProvider = Provider<GeoLocationService>(
   (ref) => GeoLocationService(
     geoLocator: GeolocatorPlatform.instance,
   ),
 );
+
+final liveLocationServiceProvider = StreamProvider<GeoPoint?>((ref) {
+  final locationService =
+      GeoLocationService(geoLocator: GeolocatorPlatform.instance);
```

Comment:
I suggest you read geoLocationServicePorivder here instead.

Reply:
Fixed with 22ce5e3